mypy/semanal.py

```diff
@@ -5329,10 +5331,33 @@ def visit_type_alias_stmt(self, s: TypeAliasStmt) -> None:
                 and isinstance(existing.node, (PlaceholderNode, TypeAlias))
                 and existing.node.line == s.line
             ):
-                existing.node = alias_node
+                updated = False
+                if isinstance(existing.node, TypeAlias):
+                    if existing.node.target != res:
+                        # Copy expansion to the existing alias, this matches how we update base classes
+                        # for a TypeInfo _in place_ if there are nested placeholders.
+                        existing.node.target = res
+                        existing.node.alias_tvars = alias_tvars
+                        # existing.node.no_args = no_args
```

Comment:
Why is this commented out? IIRC this is to support prehistoric edge case where you have
```python
A = B
B = C
C = D
```
and `D` is generic class. In such cases `A[int]` is allowed and equivalent to `D[int]`. If this works without this line then just delete it. (Or maybe this use case is explicitly prohibited with the new syntax, then I guess this is not needed either). For context, IIRC this was initially allowed for aliases like `List = list` + forward references.

Reply:
This is not supported by the new syntax. Type parameters needs to be explicit. I will delete the line.